modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/KeyValueProcessor.java

```diff
@@ -97,10 +97,14 @@ public void execute(IngestDocument document) {
         }
 
         String fieldPathPrefix = (targetField == null) ? "" : targetField + ".";
-        Arrays.stream(oldVal.split(fieldSplit))
-            .map((f) -> f.split(valueSplit, 2))
-            .filter((p) -> includeKeys == null || includeKeys.contains(p[0]))
-            .forEach((p) -> append(document, fieldPathPrefix + p[0], p[1]));
+        try {
+            Arrays.stream(oldVal.split(fieldSplit))
+                .map((f) -> f.split(valueSplit, 2))
+                .filter((p) -> includeKeys == null || includeKeys.contains(p[0]))
+                .forEach((p) -> append(document, fieldPathPrefix + p[0], p[1]));
+        } catch (IndexOutOfBoundsException e) {
+            throw new IllegalArgumentException("field [" + field + "] does not contain value_split [" + valueSplit + "]");
```

Comment:
I don't think we should catch IOBE here. We could catch some unintended IOBE and that would be bad as it would hide that.

I think instead we should do this:

```java
        Arrays.stream(oldVal.split(fieldSplit))
                .map((f) -> {
                    String[] kv = f.split(valueSplit, 2);
                    if (kv.length != 2) {
                        throw new IllegalArgumentException("field [" + field + "] does not contain value_split [" + valueSplit + "]");
                    }
                    return kv;
                })
                .filter((p) -> includeKeys == null || includeKeys.contains(p[0]))
                .forEach((p) -> append(document, fieldPathPrefix + p[0], p[1]));
```